CassetteTests/CASFileObjectQueue.m

```diff
@@ -20,21 +20,119 @@ @interface CASFileObjectQueueTests : XCTestCase
-    if (error != nil) {
-        XCTFail();
-    }
+    NSString *testFilePath = [NSTemporaryDirectory() stringByAppendingPathComponent:@"testfile"];
+
+    [[NSFileManager defaultManager] removeItemAtPath:[self defaultWritePath] error:nil];
+    [[NSFileManager defaultManager] removeItemAtPath:testFilePath error:nil];
+
+    CASFileObjectQueue<NSNumber *> *queue = [[CASFileObjectQueue alloc] initWithAbsolutePath:testFilePath
+                                                                                       error:&error];
+    XCTAssertNil(error, @"error during setup: %@", error.localizedDescription);
     self.queue = queue;
 }
 
 - (void)tearDown {
     [self.queue clear];
 }
 
+#pragma mark - tests
+
+- (void)testQueueCreationRelativePath {
+    NSString *relPath = @"testing";
+    NSArray<NSString *> *directoryPaths = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES);
+
+    [[NSFileManager defaultManager] removeItemAtPath:[NSString stringWithFormat:@"%@/%@",directoryPaths[0], relPath]
+                                               error:nil];
+    NSError *error;
+    CASFileObjectQueue<NSNumber *> *queue;
+    NSString *path;
+
+    error = nil;
+    path = relPath;
+    queue = [[CASFileObjectQueue alloc] initWithRelativePath:path error:&error];
+    XCTAssertNil(error, @"error creating queue at path: %@\n error: %@", path, error.localizedDescription);
+    [queue add:@(1)];
+
+    [[NSFileManager defaultManager] removeItemAtPath:[NSString stringWithFormat:@"%@/%@",directoryPaths[0], relPath]
+                                               error:nil];
+
+    error = nil;
+    path = [relPath stringByAppendingPathComponent:@"otherplace/test"];
+    queue = [[CASFileObjectQueue alloc] initWithRelativePath:path error:&error];
+    XCTAssertNil(error, @"error creating queue at path: %@\n error: %@", path, error.localizedDescription);
+    [queue add:@(1)];
+
+    [[NSFileManager defaultManager] removeItemAtPath:[NSString stringWithFormat:@"%@/%@",directoryPaths[0], relPath]
+                                               error:nil];
+
+    error = nil;
+    path = [relPath stringByAppendingPathComponent:@"testfile/testfile.tmp"];
+    queue = [[CASFileObjectQueue alloc] initWithRelativePath:path error:&error];
+    XCTAssertNil(error, @"error creating queue at path: %@\n error: %@", path, error.localizedDescription);
+    [queue add:@(1)];
+
+    [[NSFileManager defaultManager] removeItemAtPath:[NSString stringWithFormat:@"%@/%@",directoryPaths[0], relPath]
+                                               error:nil];
+
+    error = nil;
+    path = [relPath stringByAppendingPathComponent:@"otherplace/test/testfile.tmp"];
+    queue = [[CASFileObjectQueue alloc] initWithRelativePath:path error:&error];
+    XCTAssertNil(error, @"error creating queue at path: %@\n error: %@", path, error.localizedDescription);
+    [queue add:@(1)];
```

Comment:
What are each of these testing? If they are testing different things, make them separate unit tests. If they build off one another, add some comments.